datadog-agent.yaml

```diff
@@ -58,115 +71,246 @@ pipeline:
       tag: ${{package.version}}
       expected-commit: 0eec38de4bfcd752cc10dfa700c06dd7bcdb3f36
 
+  # Pull in a patch simplifying invoke tasks for checks
+  # Ref: https://github.com/DataDog/datadog-agent/pull/25348
+  # TODO: Remove when 7.55.0 is released
+  - uses: patch
+    with:
+      patches: 25348.patch
+
+  # Install `invoke` (build) dependencies. We ultimately package with venv so
+  # these won't leak into the package.
+  - runs: |
+      # install `invoke` and its dependencies
+      python3.11 -m pip install invoke requests toml pyyaml packaging
+
   - uses: go/bump
     with:
       deps: github.com/moby/buildkit@v0.13.1 golang.org/x/net@v0.23.0
       replaces: github.com/mholt/archiver/v3=github.com/anchore/archiver/v3@v3.5.2
 
-  # This step essentially replicates https://github.com/DataDog/datadog-agent/blob/main/omnibus/config/software/datadog-agent.rb
+  - if: ${{build.arch}} == 'aarch64'
+    runs: |
+      mkdir -p /usr/src
+      wget -O arch.deb http://deb.debian.org/debian-security/pool/updates/main/l/linux-5.10/linux-headers-${LINUX_HEADERS_VERSION}-arm64_${LINUX_KERNEL_VERSION}_arm64.deb
+      echo "a9f5b8614dc1dc123b13de540bb98283cd927a29438e0fe010c9e477b5448124 arch.deb" | sha256sum -c
```

Comment:
if you want you can use:

```
      - uses: fetch
        with:
          uri: URI
          expected-sha256: SHA256
          extract: false
```

This should simplify downloads a little bit. Note the `extract: false` key to basically fetch blobs which are not .tar.gz tarballs.